### DISCUSSION TITLE
Hide a dialog (custom component) internally

### DISCUSSION THREAD

**Discussion by AmosSzust** [quasarframework/quasar#16606] [General - Components / Directives / etc] [answered]
I have a custom dialog that is used in different pages.
```
<template>
  <q-dialog>
...
  </q-dialog>
</template>
```
In a page I have the following
```
<name-description-dialog v-model="showAddDialog" :title="addDialogTitle" :okEmit="addDialogEmit" nameValue=""
    descriptionValue="" />
```
The v-model exists on the page level and this is how I show/hide it. In the dialog I have a save button that emits an event that the page catches, process the request and if it succeeds, hides the dialog by `showAddDialog = false;`

I want to be able to close the dialog using a X button at the top but without emitting another event for the page to handle. Is there a way for the dialog to close itself?

**Comment by suleimanas** [comment, marked as answer]
you can use **v-close-popup** directive

like: `<q-btn ... v-close-popup />`
